### PR TITLE
Requiring invalid files throws error

### PR DESCRIPTION
**Pull request by Pika1998**

### CODE REVIEW

**Comment by Pika1998** [comment]
Please code review this and check why I am getting this error 
image

localhost:5000 shows this
```File .debug/ not found!```

**Comment by saurabhdaware** [comment]
hmmm weird... do you have latest upstream version of abell?

**Comment by saurabhdaware** [comment]
Oh wait you're running `abell serve` in abell-renderer repositories. To run main example you'll have to run `npm run example main` from root of abell-renderer repository.

https://github.com/abelljs/abell-renderer/blob/main/CONTRIBUTING.md

**Comment by Pika1998** [comment]
Yes that works fine :D

**Comment by Pika1998** [comment]
Actually can you merge this PR, this throws error if anything goes wrong which later can be caught in abell

By using 
```js
try{
build(...)
}catch(err)
{
Log the errror
}
```

**Comment by saurabhdaware** [comment]
Can you check why tests are failing?

**Comment by Pika1998** [comment]
> Can you check why tests are failing?

yes, i think that last test is failing because i am throwing a different error rather than the one which you had set, and now i am unable to find out what error you had thrown XD

**Comment by Pika1998** [comment]
@saurabhdaware  i noticed that you are returning not defined error in Abell instead of abell-renderer, has it got something to do with it?

**Comment by Pika1998** [comment]
```
 expect(errorStackFirstLine.trim().startsWith('execute')).to.equal(true);
```
i modified the test a bit, but the following is failing

**Comment by saurabhdaware** [comment]
> @saurabhdaware i noticed that you are returning not defined error in Abell instead of abell-renderer, has it got something to do with it?

I have to catch it in serve because I don't want the http serve to crash on that error. So now when you get undefined variable, it will show error but won't crash the whole server.

**Comment by saurabhdaware** [comment]
> ```
>  expect(errorStackFirstLine.trim().startsWith('execute')).to.equal(true);
> ```
> 
> i modified the test a bit, but the following is failing

I check if the error is being thrown from `execute` file. If we get an error, we get stack trace with it, which we can track to see where exactly the error is coming from. Is your try catch making it fail at render file? with that we won't be able to track errors down to files. 

So if we're adding try-catch, we have to add them on bottom level where exactly that file is.

**Comment by Pika1998** [comment]
Yes it is throwing an error at render () method

**Comment by Pika1998** [comment]
@saurabhdaware check now

**Comment by Pika1998** [comment]
Also, I added a new test to check if abell-renderer throws an error when user tries to require a file which does not exist.

File: src/render-utils.js [hunk 1]
```diff
@@ -41,13 +41,17 @@ function abellRequire(pathToRequire, options) {
   }
 
   const fullPathToRequire = path.join(options.basePath, pathToRequire);
-  if (fs.existsSync(fullPathToRequire)) {
-    // Local file require
-    return require(fullPathToRequire);
+  if (fullPathToRequire.includes('.')) {
```
Comment (saurabhdaware) [inline src/render-utils.js:44]: Would this cause issues if someone `require('next/links.js')`?

Instead we can
- check if the path exists
  - If exists, require the local file
  - if does not exist, try to require it as NPM package
  - If requiring NPM packages fails with module not found error, throw error that `NPM module/File does not exist`

Comment (Pika1998) [reply src/render-utils.js:44]: Interesting, that's a better method, I'll make these changes and send the PR, what you're saying is correct

**Comment by saurabhdaware** [comment]
I've got a good news and bad news :(

So by mistakely, while experimenting, I fixed this issue in next branch
Screenshot of terminal output that shows error is thrown at .abell file
This is the error message now.

In abell-renderer v0.2 we will have a feature that lets us trace the .abell file so we can point to the abell file which has error message.

The code is in the branch next.

Bad news is... since the issue is resolved... 🥺👉👈 I will have to close this PR :(

**Comment by Pika1998** [comment]
No problem, I'm glad it's solved 😁

**Pull request closed without merging**